src/Type/MediaItem/MediaItemType.php

```diff
@@ -0,0 +1,286 @@
+ *          'fields' ], 10, 1 );
+ *
+ * @package WPGraphQL\Type\MediaItem
+ */
+class MediaItemType {
+
+	/**
+	 * Holds the object definition for media details
+	 *
+	 * @var object $media_details
+	 */
+	private static $media_details;
+
+	/**
+	 * Holds the object definition for media item meta
+	 *
+	 * @var object $media_item_meta
+	 */
+	private static $media_item_meta;
+
+	/**
+	 * Holds the object definition for media sizes
+	 *
+	 * @var object $media_sizes
+	 */
+	private static $media_sizes;
+
+	/**
+	 * This customizes the fields for the mediaItem type ( attachment post_type) as the shape of the mediaItem Schema
+	 * is different than a standard post
+	 *
+	 * @see: wp-graphql.php - add_filter( 'graphql_mediaItem_fields' );add_filter( 'graphql_mediaItem_fields', [
+	 *       '\WPGraphQL\Type\MediaItem\MediaItemType', 'fields' ], 10, 1 );
+	 *
+	 * @param array $fields
+	 *
+	 * @return array
+	 */
+	public static function fields( $fields ) {
+
+		/**
+		 * Deprecate fields for the mediaItem type.
+		 * These fields can still be queried, but are just not preferred for the mediaItem type
+		 *
+		 * @since 0.0.6
+		 */
+		$fields['excerpt']['isDeprecated']      = true;
+		$fields['excerpt']['deprecationReason'] = __( 'Use the caption field instead of excerpt', 'wp-graphql' );
+		$fields['content']['isDeprecated']      = true;
+		$fields['content']['deprecationReason'] = __( 'Use the description field instead of content', 'wp-graphql' );
+
+		/**
+		 * Add new fields to the mediaItem type
+		 *
+		 * @since 0.0.6
+		 */
+		$new_fields = [
+			'caption'      => [
+				'type'        => Types::string(),
+				'description' => esc_html__( 'The caption for the resource', 'wp-graphql' ),
+				'resolve'     => function( \WP_Post $post, $args, $context, ResolveInfo $info ) {
+					$caption = apply_filters( 'the_excerpt', apply_filters( 'get_the_excerpt', $post->post_excerpt, $post ) );
+
+					return ! empty( $caption ) ? $caption : null;
+				},
+			],
+			'altText'      => [
+				'type'        => Types::string(),
+				'description' => esc_html__( 'Alternative text to display when resource is not displayed', 'wp-graphql' ),
+				'resolve'     => function( \WP_Post $post, $args, $context, ResolveInfo $info ) {
+					return get_post_meta( $post->ID, '_wp_attachment_image_alt', true );
+				},
+			],
+			'description'  => [
+				'type'        => Types::string(),
+				'description' => esc_html__( 'Description of the image (stored as post_content)', 'wp-graphql' ),
+				'resolve'     => function( \WP_Post $post, $args, $context, ResolveInfo $info ) {
+					return apply_filters( 'the_content', $post->post_content );
+				},
+			],
+			'mediaType'    => [
+				'type'        => Types::string(),
+				'description' => esc_html__( 'Type of resource', 'wp-graphql' ),
+				'resolve'     => function( \WP_Post $post, $args, $context, ResolveInfo $info ) {
+					return wp_attachment_is_image( $post->ID ) ? 'image' : 'file';
```

Comment:
I'm wondering if it would be more useful to return the mime type here, rather than just 'image' or 'file' here. We could just use `get_post_mime_type()` which pulls from the `post_mime_type` column. Thoughts?